src/sensor/ping360.cpp

```diff
@@ -80,16 +80,20 @@ void Ping360::startPreConfigurationProcess()
     writeMessage(msg);
 }
 
-void Ping360::checkBaudrateProcess()
+void Ping360::checkBaudrateProcess(bool reset)
 {
     // We use the pre configuration message to check for valid baud rates
     startPreConfigurationProcess();
 
     static int count = _ABRTotalNumberOfMessages;
 
+    if(reset) {
+        count = _ABRTotalNumberOfMessages;
+    }
+
     // We are starting to check the new baud rate
     if(count == _ABRTotalNumberOfMessages) {
-        detectBaudrates();
+        detectBaudrates(reset);
```

Comment:
should this be?

```suggestion
        detectBaudrates(true);
```

Reply:
This will reset the function each time, making it back to the highest baudrate. The idea of reset is to restart all the procedure.
The reset is only true at the first time that the detection run or after an user request.